libtsuba/src/AsyncOpGroup.cpp

```diff
@@ -0,0 +1,51 @@
+#include "tsuba/AsyncOpGroup.h"
+
+bool
+tsuba::AsyncOpGroup::FinishOne() {
+  auto op_it = pending_ops_.begin();
+  if (op_it == pending_ops_.end()) {
+    return false;
+  }
+  auto res = op_it->result.get();
+  if (!res) {
+    KATANA_LOG_DEBUG(
+        "async op for {} returned {}", op_it->location, res.error());
+    errors_++;
+    last_error_ = res.error();
+  } else {
+    res = op_it->on_complete();
+    if (!res) {
+      KATANA_LOG_DEBUG(
+          "complete cb for async op for {} returned {}", op_it->location,
+          res.error());
+    }
+  }
+  pending_ops_.erase(op_it);
```

Comment:
Why `erase` instead of `pop_front()`?

Reply:
because I already had an iterator so I figured why not 🤷 ?

Reply:
Fair enough. This just reminded me of another piece of code you wrote recently (I don't remember where) that did a pop.

Reply:
Bold of you to assume that I'm that consistent 😅